### PR TITLE
TKR conversion to v1a3: spec.kubernetes.version should start with v

### PR DESCRIPTION
**Pull request by imikushin**
#### Modify TKR conversion webhooks, to make sure that in TKR v1alpha3 `spec.version` and `spec.kubernetes.version` are prefixed with `v`.

Version string from TKR `spec.kubernetes.version` is injected into Cluster `spec.topology.version`. Without this fix, this Kubernetes version string, coming from a TKR converted from v1alpha2, may not be prefixed with `v` and will be rejected by the Cluster validation webhook.

#### Convert to/from TKR v1alpha3 `spec.osImages`

TKR v1alpha1 and v1alpha2 use `spec.nodeImageRef` to refer to the VM image to be used as the node image for the target clusters. TKR v1alpha3 uses the `spec.osImages` list to refer to available node images.

Without this change, when converting to and from TKR v1alpha3, this information was lost.

### Which issue(s) this PR fixes

N/A

### Describe testing done for PR

Updated unit tests to test conversion correctness.

### Release note

```release-note
NONE
```

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by imikushin** [comment]
Looks like I've found an issue. Looking into it.

**Comment by imikushin** [comment]
Ready for review... I think.

**Review by prkalle — APPROVED** [review]
LGTM. Thanks!

File: apis/run/v1alpha1/conversion.go [hunk 1]
```diff
@@ -22,7 +24,7 @@ func (spoke *TanzuKubernetesRelease) ConvertTo(hubRaw conversion.Hub) error {
 	hub := hubRaw.(*v1alpha3.TanzuKubernetesRelease)
 
 	if err := autoConvert_v1alpha1_TanzuKubernetesRelease_To_v1alpha3_TanzuKubernetesRelease(spoke, hub, nil); err != nil {
-		return nil
+		return err
```
Comment (imikushin) [inline apis/run/v1alpha1/conversion.go:27]: Pure luck we've never hit this.

File: apis/run/v1alpha1/conversion_test.go [hunk 2]
```diff
@@ -7,32 +7,90 @@ import (
...
+	repeat(1000, func() {
+		tkrV1a3 := &v1alpha3.TanzuKubernetesRelease{}
+		fuzzr.Fuzz(tkrV1a3)
+
+		tkrConverted := &TanzuKubernetesRelease{}
+		g.Expect(tkrConverted.ConvertFrom(tkrV1a3)).To(Succeed())
+
+		nodeImageRefs := ptrAsSliceForObjectReference(tkrConverted.Spec.NodeImageRef)
+		g.Expect(len(nodeImageRefs) == 0).To(Equal(len(tkrV1a3.Spec.OSImages) == 0),
+			"nodeImageRef should be set iff there's at least 1 osImage")
```
Comment (nehagjain15) [inline apis/run/v1alpha1/conversion_test.go:66]: nit: if

**Review by nehagjain15 — APPROVED** [review]
LGTM

**Review by anujc25 — APPROVED** [review]

**Pull request merged**